src/main/java/com/oneops/api/util/TransitionUtil.java

```diff
@@ -0,0 +1,110 @@
+package com.oneops.api.util;
+
+import com.jayway.restassured.response.Response;
+import com.jayway.restassured.specification.RequestSpecification;
+import com.oneops.api.exception.OneOpsClientAPIException;
+import com.oneops.api.resource.model.CiResource;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+public class TransitionUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TransitionUtil.class);
+
+    public static void deployAndCheckStatus(String environmentName, String transitionEnvUri, RequestSpecification request,
+                                            JSONObject jsonObject) throws OneOpsClientAPIException {
+        // Start a new deployment
+        Response response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+        checkForErrors(environmentName, response);
+
+        // Wait for 5s before checking if the deployment started successfully
+        try {
+            Thread.sleep(10000);
+        } catch(InterruptedException ie) {
+            LOG.error("Error trying to delay: {}", ie.getMessage());
+        }
+
+        // Check for capacity issues and if deployment has started or failed due to capacity error.
+        response = doDeploy(environmentName, transitionEnvUri, request, jsonObject);
+        checkForErrors(environmentName, response);
+    }
+
+    public static Response doDeploy(String environmentName, String transitionEnvUri,
+                                    RequestSpecification request, JSONObject jsonObject) throws OneOpsClientAPIException {
+
+        Response response = request.body(jsonObject.toString()).post(transitionEnvUri + environmentName + "/deployments");
```

Comment:
the main api calls for commit and deploy are all happening inside the util folder. is that normal? just curious

Reply:
Makes sense. I have moved them back to main Transition class as private methods. 
Earlier I had moved these to a separate class/file for better maintainability of the code. But now I realize that functionality-wise it does not make sense to have them in a separate class.